### ISSUE TITLE
Continuous Snowflake Login Using SSO 

### ISSUE THREAD

**Issue by daveackermanSPS** [dbeaver/dbeaver#35835]
### Description

Since upgrading to the latest version snowflake connections using SSO (externalbrowser) continuously tries to connect. Anytime anything is typed into the query editor or a query is run a new tab in the browser is opened and dbeaver connects to snowflake again.  

### DBeaver Version

24.2.1

### Operating System

macOS Sonoma 14.6.1

**Comment by rjafarau** [comment]
The same issue

**Comment by E1izabeth** [comment]
Hi, @daveackermanSPS @rjafarau 
Thanks for reporting the issue.
Does unchecking the `Open auth links in embedded browser` help? You can find this setting in `Preferences->User Interface`

**Comment by daveackermanSPS** [comment]
@E1izabeth It's currently already unchecked

**Comment by Matvey16** [comment]
@daveackermanSPS What version did you upgrade from?

**Comment by rjafarau** [comment]
Hi @E1izabeth @Matvey16 @daveackermanSPS !

In my case, the reason was an error during reading database table/columns for semantic analysis (`Settings` -> `Editors` -> `SQL Editor` -> `Code Editor` -> `Query Analysis` -> `Read database table/columns for semantic analysis`). This background task tried to parse database structure but failed to parse one database that was installed from Snowflake Marketplace. The database has a trial period. It expired and the data provider restricted access to the database. Dbeaver reconnected again and again to parse the DB structure. Let me provide an error message when I tried to view the database
![image](https://github.com/user-attachments/assets/85f29042-17b1-45b7-975b-3881c35d8510)

Oh yeah. There was one interesting moment. Originally, this database wasn’t listed in the database list. To show it I enabled `Show system objects` option (`Connection's context menu` -> `Connection view` -> `Show system objects`)
